packages/relay/src/lib/eth.ts

```diff
@@ -1220,7 +1220,7 @@ export class EthImpl implements Eth {
     
     try {
       // ETH_CALL_DEFAULT_TO_CONSENSUS_NODE = false enables the use of Mirror node
-      if ((process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE === 'undefined') || (process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE == 'false')) {
+      if ((String(process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE) === 'undefined') || (process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE == 'false')) {
```

Comment:
looks fine but can't we just test for undefined without converting to a string?

Reply:
It's a serious gotcha.  The mocha tests pass, but when stepping through it with the debugger you can see that when process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE is undefined the condition is not honored, and the js === strict equality operator checking if (process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE) === 'undefined' will not resolve to true.  It needs to be cast to a string which is the same type.

Reply:
@ebadiere, @lukelee-sl's question was spot on. The fix here should have been to remove the string format of the `'undefined'` and make it just `undefined`.


The comparison would always fail because the item on the left is not a string and the item on the right is always a string.

We should revert this change as we've added an unnecessary conversion for every single `eth_call` instance